Ticketing/Ticketing.BAL/Services/EmailService.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Microsoft.Extensions.Options;
+using SendGrid.Helpers.Mail;
+using SendGrid;
+using Ticketing.BAL.Contracts;
+using Ticketing.BAL.Options;
+using Polly;
+using Polly.Retry;
+using Ticketing.DAL.Domains;
+using Ticketing.DAL.Repositories;
+using Microsoft.Extensions.DependencyInjection;
+using static Ticketing.DAL.Enums.Statuses;
+using log4net;
+
+namespace Ticketing.BAL.Services
+{
+    public class EmailService : IEmailService
+    {
+        private readonly IOptions<EmailSettings> _optionsEmailSettings;
+        private readonly IOptions<RetryPolicySettings> _optionsRetryPolicySettings;
+        private readonly AsyncRetryPolicy<Response> _retryPolicy;
+        private readonly IServiceScopeFactory _serviceScopeFactory;
+        private readonly ILog _logger;
+
+        public EmailService(IOptions<EmailSettings> optionsEmailSettings,
+            IOptions<RetryPolicySettings> optionsRetryPolicySettings,
+            IServiceScopeFactory serviceScopeFactory,
+            ILog logger)
+        {
+            _optionsEmailSettings = optionsEmailSettings;
+            _optionsRetryPolicySettings = optionsRetryPolicySettings;
+            _serviceScopeFactory = serviceScopeFactory;
+            _logger = logger;
+
+            _retryPolicy = RetryPolicy(_optionsRetryPolicySettings.Value.RetryCount, _optionsRetryPolicySettings.Value.SleepDurationSec);
+        }
+
+        public async Task SendEmailAsync(EmailNotification emailNotification)
+        {
+            _logger.Info($"EmailService Start Send Email.");
+            EmailNotification savedEmailNotification = await SaveNotificationAsync(emailNotification);
+
+            string? fromEmail = _optionsEmailSettings.Value.SenderEmail;
+            string? fromName = _optionsEmailSettings.Value.SenderName;
+            string? apiKey = _optionsEmailSettings.Value.ApiKey;
+            var sendGridClient = new SendGridClient(apiKey);
+            var from = new EmailAddress(fromEmail, fromName);
+            var to = new EmailAddress(emailNotification.Email, emailNotification.Name);
+            var subject = "Email Notification";
+
+            var body = $"This is Email Notification order {emailNotification.OrderId} amount {emailNotification.OrderAmount} summary {emailNotification.OrderSummary}";
```

Comment:
Looks like body can be part of the EmailNotification

Reply:
EmailNotification is POCO class to write information in the repository